doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,167 +866,137 @@ Module system
+      | @module_type @module_expr_atom
+      | @module_type with @with_declaration
+      with_declaration ::= Definition @qualid {? @univ_decl } := @term
+      | Module @qualid := @qualid
+      module_expr_atom ::= @qualid
+      | ( {+ @module_expr_atom } )
+      of_module_type ::= : @module_type_inl
+      | {* <: @module_type_inl }
+      module_expr_inl ::= ! {+ @module_expr_atom }
+      | {+ @module_expr_atom } {? @functor_app_annot }
 
-   This command is used to start an interactive module named :token:`ident`.
+   This command defines a module named :token:`ident`.  See the example :ref:`here<example_def_simple_module>`.
 
-.. cmdv:: Module @ident {* @module_binding}
+   The :n:`Import` and :n:`Export` flags specify whether the module should be automatically
+   imported or exported.
 
-   Starts an interactive functor with
-   parameters given by module_bindings.
+   Specifying :n:`{* @module_binder }` starts an interactive functor with
+   parameters given by the :n:`@module_binder`\s.
 
-.. cmdv:: Module @ident : @module_type
+   .. todo: "interactive" means proof mode?
 
-   Starts an interactive module specifying its module type.
+   :n:`@of_module_type` specifies the module type.  :n:`{+ <: @module_type_inl }`
+   starts a module that satisfies each :n:`@module_type_inl`.
 
-.. cmdv:: Module @ident {* @module_binding} : @module_type
+   :n:`:= {+<+ @module_expr_inl }` specifies the body for a module or functor definition.  If it's
+   not specified, then the module is defined interactively.
+   The ! prefix indicates that any
+   `Inline` directive in the type of the functor arguments will be ignored
+   (see the :cmd:`Module Type` command below).
 
-   Starts an interactive functor with parameters given by the list of
-   :token:`module_bindings`, and output module type :token:`module_type`.
-
-.. cmdv:: Module @ident <: {+<: @module_type }
-
-   Starts an interactive module satisfying each :token:`module_type`.
-
- .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type }.
-
-   Starts an interactive functor with parameters given by the list of
-   :token:`module_binding`. The output module type
-   is verified against each :token:`module_type`.
-
-.. cmdv:: Module {| Import | Export }
-
-   Behaves like :cmd:`Module`, but automatically imports or exports the module.
-
-Reserved commands inside an interactive module
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. cmd:: Include @module
-
-   Includes the content of module in the current
-   interactive module. Here module can be a module expression or a module
-   type expression. If module is a high-order module or module type
-   expression then the system tries to instantiate module by the current
-   interactive module.
-
-.. cmd:: Include {+<+ @module}
-
-   is a shortcut for the commands :n:`Include @module` for each :token:`module`.
-
-.. cmd:: End @ident
+   .. todo: "interactively" above meaning in proof mode?  If so, maybe use that language
```

Comment:
Not really "proof mode". Proof mode changes the parser to parse tactics and this is not the case here. However, it is similar in the sense that the definition of the module is between the `Module` and the `End` keywords instead of being after the `:=`.